autogenerate_api_client.sh

```diff
@@ -0,0 +1,18 @@
+curl -X POST https://generator3.swagger.io/api/generate -H 'content-type: application/json' -d '{ "specURL" : "https://tdei-gateway-dev.azurewebsites.net/v3/api-docs", "lang" : "typescript-axios", "type" : "CLIENT", "codegenVersion" : "V3"}' --output tdei-client-axios.zip
```

Comment:
is this endpoint `https://tdei-gateway-dev.azurewebsites.net/v3/api-docs` always available? 

can you make changes to this script to take the endpoint as input?

Reply:
The api-client workflow is planned to be triggered automatically when the api gateway main/src contents is changed. Are we planing to make a change in the way the workflow is triggered?

Reply:
@santaduda the question is about the gateway API endpoint, ideally, this is what should happen.

- A Commit/PR is merged to the gateway repo
- New Code is deployed to some environment
- GitHub Workflow validates if there are any changes to the `src/main` folder or maybe, we can further refine this only to validate if there are changes to any code files ` for e.g. (.java files changed)`
- If the above validation results in a success code, it dispatches an event to auto-generate a new API client
-  On the API Client repo, a listener listens to the above event and fires up the bash script for auto client generation

- The shell script should take two inputs
```sh 
bash autogenerate_api_client.sh <ENDPOINT> <npmVersion>
```

Reply:
@ayydevara The above workflow is how it is implemented currently. 
I have updated the workflow to pass only 1 parameter <ENDPOINT> the <npmVersion> I can read it from the package.json and increment the version.

As per the github api documentation https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28 the workflow input can be passed with the "inputs" object.